File: Core/Stateflows.Common/Activities/Classes/ActivityLocator.cs
﻿using Stateflows.Activities;

namespace Stateflows.Common.Activities.Classes
{
    internal class ActivityLocator : IActivityLocator
    {
        public IBehaviorLocator Locator { get; }

        public ActivityLocator(IBehaviorLocator locator)
        {
            Locator = locator;
        }

        public bool TryLocateActivity(ActivityId id, out IActivityBehavior activity)
            => Locator.TryLocateActivity(id, out activity);
    }
}

File: Transport/@stateflows/common/src/locators/activity.locator.ts
import { Activity } from "../behaviors/activity";
import { Behavior } from "../behaviors/behavior";
import { ActivityId } from "../ids/activity.id";
import { IActivity } from "../interfaces/activity";
import { IActivityLocator } from "../interfaces/activity.locator";
import { IBehaviorLocator } from "../interfaces/behavior.locator";

export class ActivityLocator implements IActivityLocator {
    constructor(private behaviorLocator: IBehaviorLocator) {}

    locateActivity(id: ActivityId): Promise<IActivity> {
        return new Promise<IActivity>((resolve, reject) => {
            this.behaviorLocator.locateBehavior(id)
                .then(behavior => resolve(new Activity(behavior as Behavior)))
                .catch(_ => reject("State Machine not found"));
        });
    }
}